docs/src/components/Toc.tsx

```diff
@@ -2,25 +2,90 @@ import { useEffect, useRef, useState } from 'react';
 import { createPortal } from 'react-dom';
 
 import { Box, Headline, Link, List } from '@marigold/components';
+import React from 'react';
 
 export interface TocProps {
   selector: string;
   items: string;
 }
 
+function getId(children: string) {
+  return children
+    ?.split(' ')
+    .map(word => word.toLowerCase())
+    .join('-');
+}
+
+function useScrollSpy(ids: string[], options: IntersectionObserverInit) {
+  const [activeId, setActiveId] = React.useState<string>();
+  const observer = React.useRef<IntersectionObserver>();
+  React.useEffect(() => {
+    const elements = ids.map(id => document.getElementById(id));
+    observer.current?.disconnect();
+    observer.current = new IntersectionObserver(entries => {
+      entries.forEach(entry => {
+        if (entry?.isIntersecting) {
+          setActiveId(entry.target.id);
+        }
+      });
+    }, options);
+    elements.forEach(el => {
+      if (el) {
+        observer.current?.observe(el);
+      }
+    });
+    return () => observer.current?.disconnect();
+  }, [ids, options]);
+  return activeId;
+}
+
 export const Toc = ({ items, selector }: TocProps) => {
   const elements = JSON.parse(items) as { anchor: string; title: string }[];
   const [, setMounted] = useState(false);
+  const isBottom = useRef(false);
 
   const ref = useRef<Element>();
 
+  const activeId = useScrollSpy(
+    elements.map((i: { title: string }) => getId(i.title)),
+    { rootMargin: '-10% -35% -80% -15%', threshold: 0 }
+  );
+
   useEffect(() => {
     if (typeof window !== 'undefined') {
       ref.current = document.querySelector(selector) || undefined;
       setMounted(true);
     }
   }, [selector]);
 
+  useEffect(() => {
+    window.addEventListener('scroll', function () {
+      const elements = document.querySelectorAll(
+        '[data-intersection-active]'
+      ) as any;
+      const elementsLenght = elements.length;
+      const activeElement =
+        elements[elementsLenght - 2]?.dataset.intersectionActive;
+      if (
+        window.innerHeight + window.scrollY >= document.body.scrollHeight &&
+        activeElement === 'true' &&
+        isBottom.current === false
+      ) {
+        elements[elementsLenght - 2].style.fontWeight = '400';
```

Comment:
Also: can we add a class or `[data-attribute]` and do the styling inside Toc!? :)